### PR TITLE
Refactor how REST works for guilds

### PR DESCRIPTION
**Pull request by JakeMakesStuff**
# Description

Relating (but not completely fixing) #316. Note that the documentation and additional attribute specific guild stuff are not complete yet since the docs aren't ready yet and some resource specific methods I want to add require cache changes which I am waiting for upstream changes for.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [X] I ran `go generate`
- [X] I have performed a self-review of my own code (remember to run `go fmt ./...`)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] Added benchmarks if this is a performant required component (potential bottlenecks)

### CODE REVIEW

File: session.go [hunk 1]
```diff
@@ -203,174 +196,13 @@ type RESTChannel interface {
...
-	// Returns the updated emoji object on success. Fires a Guild Emojis Update Gateway event.
-	UpdateGuildEmoji(ctx context.Context, guildID, emojiID Snowflake, flags ...Flag) *updateGuildEmojiBuilder
-
-	// DeleteGuildEmoji Delete the given emoji. Requires the 'MANAGE_EMOJIS' permission. Returns 204 No Content on
-	// success. Fires a Guild Emojis Update Gateway event.
-	DeleteGuildEmoji(ctx context.Context, guildID, emojiID Snowflake, flags ...Flag) error
-}
-
 // RESTGuild REST interface for all guild endpoints
 type RESTGuild interface {
```
Comment (andersfylling) [inline session.go:200]: You can remove this, and place the methods directly into the RESTMethods interface.

File: session.go [hunk 2]
```diff
@@ -203,174 +196,13 @@ type RESTChannel interface {
...
 	// CreateGuild Create a new guild. Returns a guild object on success. Fires a Guild Create Gateway event.
 	CreateGuild(ctx context.Context, guildName string, params *CreateGuildParams, flags ...Flag) (*Guild, error)
```
Comment (andersfylling) [inline session.go:202]: Any reason you did not move the create method into the guild resource?

Comment (JakeMakesStuff) [reply session.go:202]: Yeah, that was done because .Guild needs an ID which we do not have if the guild doesn't exist.

File: session.go (hunk 1 above)
Comment (JakeMakesStuff) [reply session.go:200]: Ah yeah good spot, I'll fix soon.

**Pull request merged**